src/common/config.cc

```diff
@@ -1227,6 +1227,47 @@ void md_config_t::diff(
   }
 }
 
+void md_config_t::diff_setting(const md_config_t *other,
```

Comment:
they can be very different, if `diff` is not a `map`. and in your case, `diff` could have been a `pair<string, string>`, because the queried setting can only be matched once at most.